Realm/RLMScheduler.mm

```diff
@@ -72,8 +93,81 @@ - (void *)cacheKey {
 }
 @end
 
+namespace {
+class ActorScheduler final : public realm::util::Scheduler {
+public:
+    ActorScheduler(void (^invoke)(dispatch_block_t), dispatch_block_t verify)
+    : _invoke(invoke) , _verify(verify) {}
+
+    void invoke(realm::util::UniqueFunction<void()>&& fn) override {
+        auto ptr = fn.release();
+        _invoke(^{
+            realm::util::UniqueFunction<void()> fn(ptr);
+            fn();
+        });
+    }
+
+    bool is_on_thread() const noexcept override {
+        _verify();
+        return true;
+    }
+
+    bool is_same_as(const Scheduler *) const noexcept override {
+        return false;
```

Comment:
Is this correct? Shouldn't this check if it's the same actor?

Reply:
This function is only used by objectstore's Realm cache, which we don't use.